fix(challenge): update Statistics Calculator: Step 27 (#53196)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/65c4dc57418fd6bfc710d61d.md

---
id: 65c4dc57418fd6bfc710d61d
title: Step 29
challengeType: 0
dashedName: step-29
---

# --description--

> Add an empty `forEach` function and pass `el` as the parameter.

Reviewer: We should provide some more context for why we are looping here

```suggestion
Inside your `getMode` function, add an empty `forEach` on your `array` and pass in `el` for the parameter.

In the next few steps, you will use this loop to count the frequency of occurrences of each number in the array.
```

# --hints--

> Your getMode function should use the .forEach() method.

Reviewer: We always want to wrap functions and variables inside backticks

```suggestion
Your `getMode` function should use the `.forEach()` method.
```

```js
assert.match(getMode.toString(), /array\.forEach\(/);
```

> Your .forEach() method should have a callback function which takes an el parameter.

Reviewer: ```suggestion
Your `.forEach()` method should have a callback function which takes an `el` parameter.
```

---
id: 65c4dc57418fd6bfc710d61d
title: Step 29
challengeType: 0
dashedName: step-29
---

# --description--

> Inside your `getMode` function, add a `forEach` on your `array` parameter. Your `.forEach()` method should have an empty callback function which takes an `el` parameter.

Reviewer 2: ```suggestion
Inside your `getMode` function, chain a `forEach` to the returned `array` parameter. Your `.forEach()` method should have an empty callback function which takes an `el` parameter.
```
